### PR TITLE
docs: fix exclude option for quickstart examples.

### PR DESCRIPTION
**Pull request by marcosnils**
this adds the exclude option to the proper `client.Host().Directory`
operation to avoid sending the files over the wire each time

Signed-off-by: Marcos Lilljedahl <marcosnils@gmail.com>

### CODE REVIEW

**Comment by vikram-dagger** [comment]
@marcosnils I believe this needs further changes:
- Text in "test the application" page should change to "In this case, it excludes the node_modules (locally-installed dependencies), ci (pipeline code) and build (build artifacts) directories."
- All the playground embeds should be regenerated as otherwise they will reflect the previous code.

**Comment by marcosnils** [comment]
> * All the playground embeds should be regenerated as otherwise they will reflect the previous code.

correct. I'll update them tomorrow :pray: 


> * Text in "test the application" page should change to "In this case, it excludes the node_modules (locally-installed dependencies), ci (pipeline code) and build (build artifacts) directories."

:+1:  I'll update that as well

File: docs/current/quickstart/947391-test.mdx [hunk 1]
```diff
@@ -40,7 +40,7 @@ This code listing does the following:
 - It creates a Dagger client with `Connect()` as before.
 - It uses the client's `Container().From()` method to initialize a new container from a base image - again, the `node:16-slim` image. This base image is the Node.js version to use for testing. The `From()` method returns a new `Container` object with the result.
 - It uses the `Container.WithDirectory()` method to mount the source code directory on the host at the `/src` mount point in the container, and the `Container.WithWorkdir()` method to set the working directory to that mount point.
-  - Notice that the `Container.WithDirectory()` accepts additional options to exclude (or include) specific files from the mount. In this case, it excludes the `node_modules` (locally-installed dependencies) and `ci` (pipeline code) directories.
+  - Notice that the `Container.WithDirectory()` accepts additional options to exclude (or include) specific files from the mount. In this case, it excludes the `node_modules` (locally-installed dependencies), `ci` (pipeline code) and build (build artifacts) irectories.
```
Comment (wingyplus) [inline docs/current/quickstart/947391-test.mdx:43]: Is `d` in `irectories` is missing?

```suggestion
  - Notice that the `Container.WithDirectory()` accepts additional options to exclude (or include) specific files from the mount. In this case, it excludes the `node_modules` (locally-installed dependencies), `ci` (pipeline code) and build (build artifacts) directories.
```

Comment (vikram-dagger) [inline docs/current/quickstart/947391-test.mdx:43]:
```suggestion
  - Notice that the `Container.WithDirectory()` accepts additional options to exclude (or include) specific files from the mount. In this case, it excludes the `node_modules` (locally-installed dependencies), `ci` (pipeline code) and `build` (build artifacts) directories.
```

Comment (vikram-dagger) [reply docs/current/quickstart/947391-test.mdx:43]: @marcosnils this text should also be adjusted for the other two tabs in this page.

Comment (vikram-dagger) [reply docs/current/quickstart/947391-test.mdx]: @marcosnils I think you may have missed my comment above

**Comment by vikram-dagger** [comment]
@marcosnils the updated embeds do not appear in the deploy preview, the older embeds appear instead. 
image

**Comment by TomChv** [comment]
@vikram-dagger @marcosnils This PR is important to be merged as soon as possible, we had a recent issue opened #5902 related to that one.
How can I help to make it move forward? The embeds still not appears in the deploy preview, how can we fix that?

**Comment by vikram-dagger** [comment]
After redeploying the preview with a clean cache, the updated embeds appear although with errors (see below). The revised preview is at https://6530eff592401f0255ee010e--devel-docs-dagger-io.netlify.app/

Errors seen in preview:

1. In the section on multi-stage build, the Python embed shows graphql at the top.
image

2. The embed code is not updated for the section "Build the application" for all three code tabs. 
image

**Comment by marcosnils** [comment]
> 1. The text explanation needs to be changed in the remaining two tabs. See comment [docs: fix exclude option for quickstart examples. #5854 (comment)](https://github.com/dagger/dagger/pull/5854#discussion_r1353032126)

done.



> 2\. The updated embeds are not appearing in the deploy preview. I am not sure if this is an issue with the playground or netlify caching. See comment [docs: fix exclude option for quickstart examples. #5854 (comment)](https://github.com/dagger/dagger/pull/5854#issuecomment-1759337802). To check, I am redeploying the preview using a clean cache on netlify: https://app.netlify.com/sites/devel-docs-dagger-io/deploys/6530eff592401f0255ee010e

checking.

**Comment by marcosnils** [comment]
done @vikram-dagger. This is ready for a final pass.

**Comment by vikram-dagger** [comment]
@marcosnils I redeployed with a clean cache and rechecked, but I continue to see the errors listed in https://github.com/dagger/dagger/pull/5854#issuecomment-1770384509

**Comment by TomChv** [comment]
I can still see it partially updated, it looks ok it `test the application` but not in `build the application`

Screenshot 2023-10-30 at 12 55 11

Screenshot 2023-10-30 at 12 55 52

Same error as @vikram-dagger for `perform a multi-stage build`.

**Comment by marcosnils** [comment]
@TomChv @vikram-dagger  mind giving one last :eyes: please? :pray:

**Comment by vikram-dagger** [comment]
> @TomChv @vikram-dagger mind giving one last 👀 please? 🙏

We're almost there!

In "Build the application", the Go snippet doesn't exclude the `build` dir. Node.js and Python snippets do.

image

**Comment by marcosnils** [comment]
> In "Build the application", the Go snippet doesn't exclude the `build` dir. Node.js and Python snippets do.

oh, that's quite strange. Let me fix that really quick

**Comment by marcosnils** [comment]
> > In "Build the application", the Go snippet doesn't exclude the `build` dir. Node.js and Python snippets do.
> 
> oh, that's quite strange. Let me fix that really quick

yep, I missed that effectively. Good catch Vikram!

**Comment by marcosnils** [comment]
Done! hopefully this is the last pass :pray:

**Review by vikram-dagger — APPROVED** [review]
Looks good. However please fix the CI errors before merging @marcosnils

**Comment by marcosnils** [comment]
> Looks good. However please fix the CI errors before merging @marcosnils

Done Vikram! the scan engine CI issues are unrelated to this PR .

**Comment by vikram-dagger** [comment]
@marcosnils I was about to merge this but it seems there are various unresolved conflicts with the IDs. I'm not sure which are the correct IDs to use when resolving the conflicts, do you want to take a look?

File: docs/current/quickstart/947391-test.mdx [hunk 2]
```diff
@@ -58,14 +58,14 @@ The `From()`, `WithDirectory()`, `WithWorkdir()` and `WithExec()` methods all re
...
-- It uses the `Container.withDirectory()` method to write the host directory to the `/src` path in the container, and the `Container.withWorkdir()` method to set the working directory to that filesystem location.
-  - Notice that the `Container.withDirectory()` accepts additional options to exclude (or include) specific files from the mount. In this case, it excludes the `node_modules` (locally-installed dependencies) and `ci` (pipeline code) directories.
+- It uses the `Container.withDirectory()` method to mount the source code directory on the host at the `/src` mount point in the container, and the `Container.withWorkdir()` method to set the working directory to that mount point.
+  - Notice that the `Container.WithDirectory()` accepts additional options to exclude (or include) specific files from the mount. In this case, it excludes the `node_modules` (locally-installed dependencies), `ci` (pipeline code) and `build` (build artifacts) directories.
```
Comment (helderco) [inline docs/current/quickstart/947391-test.mdx:68]:
```suggestion
  - Notice that the `Container.withDirectory()` accepts additional options to exclude (or include) specific files from the mount. In this case, it excludes the `node_modules` (locally-installed dependencies), `ci` (pipeline code) and `build` (build artifacts) directories.
```

File: docs/current/quickstart/947391-test.mdx [hunk 3]
```diff
@@ -83,14 +83,14 @@ The `from()`, `withDirectory()`, `withWorkdir()` and `withExec()` methods all re
...
-<Embed id="2XgPeWiQsRX" />
+<Embed id="yZj9P2N03-D" />
 
 This code listing does the following:
 
 - It creates a Dagger client with `with dagger.Connection()` as before.
 - It uses the client's `container().from_()` method to initialize a new container from a base image - again, the `node:16-slim` image. This base image is the Node.js version to use for testing. The `from_()` method returns a new `Container` object with the result.
 - It uses the `Container.with_directory()` method to mount the source code directory on the host at the `/src` mount point in the container, and the `Container.with_workdir()` method to set the working directory to that mount point.
-  - Notice that the `Container.with_directory()` accepts additional options to exclude (or include) specific files from the mount. In this case, it excludes the `node_modules` (locally-installed dependencies) and `ci` (pipeline code) directories.
+  - Notice that the `Container.WithDirectory()` accepts additional options to exclude (or include) specific files from the mount. In this case, it excludes the `node_modules` (locally-installed dependencies), `ci` (pipeline code) and `build` (build artifacts) directories.
```
Comment (helderco) [inline docs/current/quickstart/947391-test.mdx:93]:
```suggestion
  - Notice that the `Container.with_directory()` accepts additional options to exclude (or include) specific files from the mount. In this case, it excludes the `node_modules` (locally-installed dependencies), `ci` (pipeline code) and `build` (build artifacts) directories.
```

File: docs/current/quickstart/947391-test.mdx (hunk 2 above)
Comment (vikram-dagger) [reply docs/current/quickstart/947391-test.mdx:68]: Thank you! https://github.com/dagger/dagger/pull/6142

File: docs/current/quickstart/947391-test.mdx (hunk 3 above)
Comment (vikram-dagger) [reply docs/current/quickstart/947391-test.mdx:93]: Thank you! https://github.com/dagger/dagger/pull/6142

**Pull request merged**